staging/vhost-device-sound/src/lib.rs

```diff
@@ -1,5 +1,34 @@
 // Manos Pitsidianakis <manos.pitsidianakis@linaro.org>
 // SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+//
+#![deny(
+    /* groups */
+    clippy::correctness,
+    clippy::suspicious,
+    clippy::complexity,
+    clippy::perf,
+    clippy::style,
+    clippy::nursery,
+    //* restriction */
+    clippy::dbg_macro,
+    clippy::rc_buffer,
+    clippy::as_underscore,
+    clippy::assertions_on_result_states,
+    //* pedantic */
+    clippy::cast_lossless,
+    clippy::cast_possible_wrap,
+    clippy::ptr_as_ptr,
+    clippy::bool_to_int_with_if,
+    clippy::borrow_as_ptr,
+    clippy::case_sensitive_file_extension_comparisons,
+    clippy::cast_lossless,
+    clippy::cast_ptr_alignment,
+    clippy::naive_bytecount
+)]
+#![allow(
+    clippy::significant_drop_in_scrutinee,
+    clippy::significant_drop_tightening
+)]
```

Comment:
Is it possible to enable these in some way for the entire workspace?

Reply:
Out of curiosity, on what basis did we decide which ones to enable?

Reply:
I took the basic categories from here:

https://doc.rust-lang.org/nightly/clippy/

```text
clippy::correctness	code that is outright wrong or useless	deny
clippy::suspicious	code that is most likely wrong or useless	warn
clippy::complexity	code that does something simple but in a complex way	warn
clippy::perf	code that can be written to run faster	warn
clippy::style	code that should be written in a more idiomatic way	warn
clippy::pedantic	lints which are rather strict or might have false positives	allow
clippy::nursery	new lints that are still under development	allow
clippy::cargo	lints for the cargo manifest	allow
```

And enabled all of them, then disabled `pedantic` and only cherry picked some of them.

For context, pedantic has some... really pedantic lints like:

```text
error: item name starts with its containing module's name
   --> vhost-device-sound/src/virtio_sound.rs:269:12
    |
269 | pub struct VirtioSoundPcmHeader {
    |            ^^^^^^^^^^^^^^^^^^^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_name_repetitions
```

Also no wildcard imports, etc.

Reply:
> Is it possible to enable these in some way for the entire workspace?

Good question. It's gonna be possible in stable soon:

https://rust-lang.github.io/rfcs/3389-manifest-lint.html

Quoting the RFC:

```text
Currently, you can configure lints through

    #[<level>(<lint>)] or #![<level>(<lint>)], like #[forbid(unsafe)]
        But this doesn't scale up with additional targets (benches, examples, tests) or workspaces
    On the command line, like cargo clippy -- --forbid unsafe
        This puts the burden on the caller
    Through RUSTFLAGS, like RUSTFLAGS=--forbid=unsafe cargo clippy
        This puts the burden on the caller
    In .cargo/config.toml's target.*.rustflags
        This couples you to the running in specific directories and not running in the right directory causes rebuilds
        The cargo team has previously stated that [they would like to see package-specific config moved to manifests](https://internals.rust-lang.org/t/proposal-move-some-cargo-config-settings-to-cargo-toml/13336/14?u=epage)
```

So to enable them for the workspace we could put them in RUSTFLAGS for all targets in a `.cargo/config.toml`.

Reply:
> I took the basic categories from here:
> 
> https://doc.rust-lang.org/nightly/clippy/
> 

Cool, thanks for sharing!
I'm thinking about whether we should add a comment here with this link, but maybe not.

> 
> And enabled all of them, then disabled `pedantic` and only cherry picked some of them.
> 
> For context, pedantic has some... really pedantic lints like:

Yeah, I agree to skip it.

Reply:
> > Is it possible to enable these in some way for the entire workspace?
> 
> Good question. It's gonna be possible in stable soon:
> 
> https://rust-lang.github.io/rfcs/3389-manifest-lint.html
> 
> Quoting the RFC:
> 
> ```
> Currently, you can configure lints through
> 
>     #[<level>(<lint>)] or #![<level>(<lint>)], like #[forbid(unsafe)]
>         But this doesn't scale up with additional targets (benches, examples, tests) or workspaces
>     On the command line, like cargo clippy -- --forbid unsafe
>         This puts the burden on the caller
>     Through RUSTFLAGS, like RUSTFLAGS=--forbid=unsafe cargo clippy
>         This puts the burden on the caller
>     In .cargo/config.toml's target.*.rustflags
>         This couples you to the running in specific directories and not running in the right directory causes rebuilds
>         The cargo team has previously stated that [they would like to see package-specific config moved to manifests](https://internals.rust-lang.org/t/proposal-move-some-cargo-config-settings-to-cargo-toml/13336/14?u=epage)
> ```
> 
> So to enable them for the workspace we could put them in RUSTFLAGS for all targets in a `.cargo/config.toml`.

Cool, can you open a PR with these details, so we don't forget to do it when it's in stable?

Reply:
> Cool, can you open a PR with these details, so we don't forget to do it when it's in stable?

An issue you mean?

Reply:
> > Cool, can you open a PR with these details, so we don't forget to do it when it's in stable?
> 
> An issue you mean?

Yep, sorry! Yesterday was a public holiday, so today is like Monday for me ;-P